spec/controllers/followings_controller_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'spec_helper'
+
+describe FollowingsController do
+  before(:each) do
+    request.env["HTTP_REFERER"] = "previous_page"
+  end
+  describe "GET people" do
+    context "with authenticated users" do
+      let(:following) { Fabricate(:following, follower_id: current_user.id) }
```

Comment:
`before` blocks are run right before the `it` block. That means you can combine the two yourself as well.  For example, you could rewrite this context like this:

```ruby
context "with authenticated users" do
  it "sets @followings" do
    request.env["HTTP_REFERER"] = "previous_page"
    following = Fabricate(:following, follower_id: current_user.id)
    set_current_user
    get :people
    expect(assigns(:followings)).to eq([following])
  end
end
```

Here is an example of why before can be a bit hard to follow when you are reading code. I missed that first line (setting the referrer) until I re-read the question before posting this reply. I had all of the other lines, but didn't even see that one.